### ISSUE TITLE
Add pages for new pub commands

### ISSUE THREAD

**Issue by kwalrath** [dart-lang/site-www#2898]
Also, commands -> subcommands.

* Subcommands we should probably document:
  * New subcommands: add, login, remove
  * Other subcommands: logout
* Subcommands we should remove: run

2.10 stable's output from `dart help pub`:
```
...
Available commands:
  cache       Work with the system cache.
  deps        Print package dependencies.
  downgrade   Downgrade the current package's dependencies to oldest versions.
  get         Get the current package's dependencies.
  global      Work with global packages.
  logout      Log out of pub.dartlang.org.
  outdated    Analyze your dependencies to find which ones can be upgraded.
  publish     Publish the current package to pub.dartlang.org.
  run         Run an executable from a package.
  upgrade     Upgrade the current package's dependencies to latest versions.
  uploader    Manage uploaders for a package on pub.dartlang.org.
  version     Print pub version.
```

2.12.0-133.7.beta output from the same:
```
...
Available subcommands:
  add         Add a dependency to pubspec.yaml.
  cache       Work with the system cache.
  deps        Print package dependencies.
  downgrade   Downgrade the current package's dependencies to oldest versions.
  get         Get the current package's dependencies.
  global      Work with global packages.
  login       Log into pub.dev.
  logout      Log out of pub.dev.
  outdated    Analyze your dependencies to find which ones can be upgraded.
  publish     Publish the current package to pub.dartlang.org.
  remove      Removes a dependency from the current package.
  upgrade     Upgrade the current package's dependencies to latest versions.
  uploader    Manage uploaders for a package on pub.dartlang.org.
```

**Comment by jonasfj** [comment]
We should also document:
 * `dart pub upgrade --major-versions`
   * This will do a resolution of all dependencies without the upper-bound constraint.
   * Then it'll update the `pubspec.yaml` with new constraints as needed.
   * TL;DR: This gives you the same versions as is listed in "Resolvable" column of `dart pub outdated`
   * Using `dart pub upgrade --major-versions foo bar` will only attempt to upgrade `package:foo` and `package:bar` while retaining the constraints for all other dependencies.
   * `dart pub upgrade --major-versions --dry-run` is a good trick to see what changes it plans to make -- it's also good advice to commit `pubspec.yaml` before using `dart pub upgrade --major-versions`, otherwise you can't undo.
 * `dart pub upgrade --null-safety`
   * This is similar to `--major-versions` except it tries to force dependencies to be using a null-safety version.
   * This will get you the "Resolvable" column from `dart pub outdated --mode=null-safety`
   * This will also guide you to do partial null-safety migrations, if you want to attempt this.
